Phil - update of code base and description

theGreatWhiteShark opened the pull request:
The code had a number of issues since it was relying on older versions of various packages (e.g. chainer). 

In addition I updated the descriptions in the README and created a troubleshooting tutorial in case someone has problems enabling the webcam.

Also there was an issue with the generated artworks. The header of the JPEGs was not set properly and I had to add a line specifying the images DPI in order for pdflatex to work with them

theGreatWhiteShark committed:
Ignoring Emacs and Cmake files

theGreatWhiteShark committed:
Adding installation, running, and troubleshooting to README

theGreatWhiteShark committed:
Adding a C++ example script testing cam with OpenCV

theGreatWhiteShark committed:
Resolving cam issue and updating troubeshooting

theGreatWhiteShark committed:
note howto change camera input device

theGreatWhiteShark committed:
chainer: remove test argument in normalization
not supported anymore in my chainer version

theGreatWhiteShark committed:
chainer: removing typo and adding comments

theGreatWhiteShark committed:
removing all 'test' heritage

theGreatWhiteShark committed:
Remove Monet image since no corr. style

theGreatWhiteShark committed:
Change default cam to USB cam

theGreatWhiteShark committed:
Ignore the log file and artwork generate during style transfer

theGreatWhiteShark committed:
Saves transformed picture after 'Kunstwerk malen' press

as 'artwork.jpg' in the folder the program is running in. This is
required for the generation of the postcard and a convenience for the user.

theGreatWhiteShark committed:
Bugfix: snapshot and transformation for smaller window sizes

Whenever the wxArt window was quite small the width of the taken
photograph was just 1 causing the transformation and all the following
steps to fail. Now the program just uses the plain picture taken from
the camera feed in those cases.

theGreatWhiteShark committed:
Updated README: installation of the package

theGreatWhiteShark committed:
Ignoring the files compiled by LaTeX

theGreatWhiteShark committed:
Ignoring further tex files from the postcard generation

theGreatWhiteShark committed:
Adding further requirements for postcard generation to README

theGreatWhiteShark committed:
Bugfix: renaming page pdfs generate by pdftk

In order to print four postcards at once, the program pdftk is used to
split default.pdf into its individual pages. Which are two in
number. Nevertheless in the script pg_0002.pdf and pg_0003.pdf were
hard coded. Since the later one didn't exist, I changed it to
pg_0001.pdf and pg_0002.pdf and everything works fine

theGreatWhiteShark committed:
Postcard: fixing cleanup

theGreatWhiteShark committed:
Bugfix: Correcting DPI information in JPEG header
 # The OpenCV package seems to not set the resolution within
        # the header of the JPEGs right. Therefore PDFLaTeX doesn't
        # know the DPI and the corresponding unit and considers the
        # e.g. 300px x 300px as 300in x 300in, which would be way
        # to big. Therefore it does include the image in the pdf
        # (you can still extract it using pdfimages -j default.pdf
        # extracts-) but does not display it.
        # Therefore we have to add the correct DPI to the header
        # ourselves.

theGreatWhiteShark committed:
Remove blank first page from PDF

theGreatWhiteShark committed:
update description on postcard

theGreatWhiteShark committed:
Update README dependencies

theGreatWhiteShark committed:
updating email text for Lange Nacht

theGreatWhiteShark committed:
Merge branch 'phil' of github.com:jusjusjus/wxArt into phil

theGreatWhiteShark committed:
Example pictures for README

theGreatWhiteShark committed:
ignoring aux file of exiftool

theGreatWhiteShark committed:
Adding description of app to README

theGreatWhiteShark committed:
README repair broken link

jusjusjus commented:
Hi, so far unable to run the updated code base.  Apparently something with backwards compatibility.
`HIGHGUI ERROR: V4L: index 1 is not correct!`
The gui shows, but opencv is unable to open the video stream.

bluens commented:
I can confirm this bug.

theGreatWhiteShark commented:
Alright, in line 98 of wxArt/Camera.py is still 

```{python}
self.capture = cv2.VideoCapture(1)
```

the second video device (an external web cam provided by the institute) selected (/dev/video1 instead of /dev/video0). 

Maybe we should introduce a new input argument so the user can select the cam.

theGreatWhiteShark committed:
Resetting default video input to /dev/video0

jusjusjus committed:
Adding logging to emailctrl.

Option -d to main flips the switch to logging.DEBUG (logging.INFO is
default).

jusjusjus commented:
Cool, video funktioniert jetzt.  E-mail hab ich nicht zum laufen bekommen.  Kann das jemand ausprobieren?

theGreatWhiteShark commented:
Die Email hat bei mir gestern einwandfrei funktioniert. Funzt auch auf dem SuSe Gerät, das ich gerade vor mir hab.

theGreatWhiteShark committed:
added installation description for Suse Linux

theGreatWhiteShark committed:
Merge branch 'phil' of github.com:jusjusjus/wxArt into phil

jusjusjus commented:
Hi Phil,

dann war das wohl mein e-mail server, der grad nicht funktioniert hat.  Wollte nur checken.

Also, meinethalben geht das jetzt klar.  Leider kann nur bene reviewen ..

jusjusjus commented:
Wenn Du Dir die Mühe machen willst, könntest Du ja sogar noch eins drauf setzen, und einen squash der verschiedenen mini-commits machen :)

theGreatWhiteShark commented:
Ach, den Sinn hinterm squashen hab ich noch nicht so wirklich verstanden. So ists doch viel detailierter :) 

Ich schau mir das Ganze nochmal kurz unter Windows an.

jusjusjus commented:
ach komm
<https://softwareengineering.stackexchange.com/questions/263164/why-squash-git-commits-for-pull-requests>
:)

Am 12. Juni 2017 um 16:24 schrieb theGreatWhiteShark <
notifications@github.com>:

> Ach, den Sinn hinterm squashen hab ich noch nicht so wirklich verstanden.
> So ists doch viel detailierter :)
>
> Ich schau mir das Ganze nochmal kurz unter Windows an.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jusjusjus/wxArt/pull/27#issuecomment-307804875>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGAdFAirRf93U4sjbger8YM3pdWmWAvYks5sDUoQgaJpZM4N2dpF>
> .
>



-- 
Justus Schwabedal

Handy (D): +49 177 939 5281
email: jschwabedal@googlemail.com
skype: justus1802

Görlitzer Str. 22
01099 Dresden, Sachsen
Germany

Steinkreuzstr. 23
53757 Sankt Augustin, NRW
Germany

theGreatWhiteShark commented:
Pfft. Also ich habe mir das Ganze gerade mal unter Windows angeschaut (auf einem der Institutslaptops). 

Man muss 1. Python herunterladen und installieren, 2. Git herunterladen und installieren, 3. LAPACK herunterladen und compilieren, 4. OpenCV herunterladen und compilieren ...

Das kann man echt niemanden zumuten der keine Ahnung von Computern hat. Da werde ich keine Beschreibung zu ins README packen.

theGreatWhiteShark committed:
Introducing input argument v for video device
By calling the script with the flag -v the user now can specify which video device to use

theGreatWhiteShark committed:
merging conflict

theGreatWhiteShark commented:
Hab noch schnell ein Inputargument für das Video device hinzugefügt.

Zu dem Squashen: es sind halt höchstens so 4 konsekutive commits, die ich squashen könnte. Hab an vielen klitzekleinen Baustellen gearbeitet.

jusjusjus committed:
Added logging to other modules.

This commit also adds an error message when starting under python3.
GIFAnimatedCtrl needs to be patched for this version.  Also, cv2
is not easily accessible under python3.

theGreatWhiteShark committed:
Adding an argument to print specify printer
Using this input flag the postcard will be send to the printer automatically and evince will not show the PDF.

theGreatWhiteShark committed:
?

Merge branch 'phil' of github.com:jusjusjus/wxArt into phil

theGreatWhiteShark commented:
@jusjusjus Kannst du dir mal den überarbeiten __init__ der Postcard Klasse anschauen? Mein Python ist erst auf Yoda-Level.

jusjusjus committed:
Refactored init of Postcard.py.

theGreatWhiteShark commented:
@jusjusjus da lief was schief. Jetzt findet er resources/postcards/default.tex nicht mehr

jusjusjus committed:
Fixed faulty join in default.tex

jusjusjus commented:
Ja, einmal os.path.join, immer os.path.join.  Ich hab jetzt nochmal umgeschrieben.  Vielleicht muss man das woanders auch noch machen.  Schau mal. Jus

theGreatWhiteShark commented:
passt

bluens commented:
I get a latex error:

! Package pdfpages Error: Cannot find file `./to_print_4x4.pdf'.
See the pdfpages package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.7 ...e,noautoscale,angle=90]{./to_print_4x4.pdf}

theGreatWhiteShark commented:
Hast du die neuste Version gepullt? Bei mir läuft es durch.

Und hast du auch die Dependencies installiert? (Siehe Readme)

bluens commented:
Here the full output:

```
INFO:ChainerFNS:Elapsed time: 3.45
sh: 1: exiftool: not found
This is pdfTeX, Version 3.14159265-2.6-1.40.16 (TeX Live 2015/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode

(/home/bluens/Documents/projects/wxArt/wxArt/../resources/postcards/default.tex
LaTeX2e <2016/02/01>
Babel <3.9q> and hyphenation patterns for 8 language(s) loaded.
(/usr/share/texlive/texmf-dist/tex/latex/koma-script/scrartcl.cls
Document Class: scrartcl 2015/10/03 v3.19a KOMA-Script document class (article)

(/usr/share/texlive/texmf-dist/tex/latex/koma-script/scrkbase.sty
(/usr/share/texlive/texmf-dist/tex/latex/koma-script/scrbase.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texlive/texmf-dist/tex/latex/koma-script/scrlfile.sty
Package scrlfile, 2015/10/03 v3.19a KOMA-Script package (loading files)
                  Copyright (C) Markus Kohm

))) (/usr/share/texlive/texmf-dist/tex/latex/koma-script/tocbasic.sty)

Package scrartcl Warning: You've used obsolete option `7pt'.
(scrartcl)                Usage of this option indicates an old document
(scrartcl)                and changes compatibility level using
(scrartcl)                `fontsize=7pt,version=first'.
(scrartcl)                If you don't want this, you should simply
(scrartcl)                replace option `7pt' by `fontsize=7pt'.

(/usr/share/texlive/texmf-dist/tex/latex/koma-script/typearea.sty
Package typearea, 2015/10/03 v3.19a KOMA-Script package (type area)
                  Copyright (C) Frank Neukam, 1992-1994
                  Copyright (C) Markus Kohm, 1994-


Package typearea Warning: DIV for 7.0pt and used papersize
(typearea)                not defined!
(typearea)                Using DIV=1.

DIV calculation for typearea with good linewidth.
)) (/usr/share/texlive/texmf-dist/tex/generic/babel/babel.sty
(/usr/share/texlive/texmf-dist/tex/generic/babel-german/ngerman.ldf
(/usr/share/texlive/texmf-dist/tex/generic/babel-german/ngermanb.ldf
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.def))))
(/usr/share/texlive/texmf-dist/tex/latex/base/inputenc.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/utf8.def
(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.dfu)
(/usr/share/texlive/texmf-dist/tex/latex/base/ot1enc.dfu)
(/usr/share/texlive/texmf-dist/tex/latex/base/omsenc.dfu)))
(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))
(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty)
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/auxhook.sty)
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty)
(/usr/share/texlive/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/hyperref.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty))

Package hyperref Message: Driver (autodetected): hpdftex.

(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hpdftex.def
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))
(/usr/share/texlive/texmf-dist/tex/latex/tools/multicol.sty)
(/usr/share/texlive/texmf-dist/tex/latex/psnfss/helvet.sty)
(/usr/share/texlive/texmf-dist/tex/latex/geometry/geometry.sty)

Class scrartcl Warning: Usage of package `parskip' together
(scrartcl)              with a KOMA-Script class is not recommended.
(scrartcl)              I'd suggest to use option
(scrartcl)              `parskip' with one of it's several values.
(scrartcl)              Nevertheless, using requested
(scrartcl)              package `parskip' on input line 12.

(/usr/share/texlive/texmf-dist/tex/latex/parskip/parskip.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/graphics.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/pdftex-def/pdftex.def)))
(./default.aux) (/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1phv.fd)
(/usr/share/texlive/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))
(./default.out) (./default.out)
*geometry* driver: auto-detecting
*geometry* detected driver: pdftex
(/usr/share/texlive/texmf-dist/tex/context/base/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty)
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))

pdfTeX warning: pdflatex: arithmetic: number too big


pdfTeX warning: pdflatex: arithmetic: number too big
<./artwork.jpg, id=4, --32768.0pt x 0.0pt> <use ./artwork.jpg>
Underfull \hbox (badness 10000) in paragraph at lines 33--37

[2{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}] (./default.aux) )
(see the transcript file for additional information){/usr/share/texlive/texmf-d
ist/fonts/enc/dvips/base/8r.enc}</usr/share/texlive/texmf-dist/fonts/type1/urw/
helvetic/uhvr8a.pfb>
Output written on default.pdf (1 page, 8711 bytes).
Transcript written on default.log.
Error: Unable to find file.
Error: Failed to open PDF file: 
   pg_0002.pdf
Error: Unable to find file.
Error: Failed to open PDF file: 
   pg_0002.pdf
Error: Unable to find file.
Error: Failed to open PDF file: 
   pg_0002.pdf
Error: Unable to find file.
Error: Failed to open PDF file: 
   pg_0002.pdf
Errors encountered.  No output created.
Done.  Input errors, so no output created.
Error: Unable to find file.
Error: Failed to open PDF file: 
   page_2.pdf
Errors encountered.  No output created.
Done.  Input errors, so no output created.
This is pdfTeX, Version 3.14159265-2.6-1.40.16 (TeX Live 2015/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(/home/bluens/Documents/projects/wxArt/wxArt/../resources/postcards/merger.tex
LaTeX2e <2016/02/01>
Babel <3.9q> and hyphenation patterns for 8 language(s) loaded.
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.sty
(/usr/share/texlive/texmf-dist/tex/generic/babel-english/english.ldf
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.def)))
(/usr/share/texlive/texmf-dist/tex/latex/geometry/geometry.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifpdf.sty)
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifvtex.sty)
(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty))
(/usr/share/texlive/texmf-dist/tex/latex/pdfpages/pdfpages.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texlive/texmf-dist/tex/latex/tools/calc.sty)
(/usr/share/texlive/texmf-dist/tex/latex/eso-pic/eso-pic.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/atbegshi.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/infwarerr.sty)
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ltxcmds.sty))
(/usr/share/texlive/texmf-dist/tex/latex/xcolor/xcolor.sty
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/color.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/pdftex-def/pdftex.def)))
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/graphics.cfg)))
(/usr/share/texlive/texmf-dist/tex/latex/pdfpages/pppdftex.def)) (./merger.aux)
*geometry* driver: auto-detecting
*geometry* detected driver: pdftex
(/usr/share/texlive/texmf-dist/tex/context/base/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texlive/texmf-dist/tex/generic/oberdiek/pdftexcmds.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifluatex.sty))
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/kvdefinekeys.sty))
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/etexcmds.sty)))
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/pdflscape.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/lscape.sty))

! Package pdfpages Error: Cannot find file `./to_print_4x4.pdf'.

See the pdfpages package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.7 ...e,noautoscale,angle=90]{./to_print_4x4.pdf}
                                                  %
? 
```

theGreatWhiteShark commented:
> sh: 1: exiftool: not found
Daher kann er den JPEG Header nicht anpassen.

> Error: Failed to open PDF file: 
>   pg_0002.pdf
Das kommt daher, dass du das pdftk Paket noch nicht installiert hast und das ursprüngliche PDF nicht in die einzelnen Seiten gesplitet werden kann.

Installieren mal bitte die Requirements aus dem README und versuche es nochmal

bluens commented:
As far as I can see it works fine.
Some small bugs remain. Hard to reproduce. Nothing critical.
Nice work, it's worthy of master.

bluens reviewed (approved):
Good work. Worthy of master.
